crates/iceberg/src/spec/table_metadata.rs

```diff
@@ -1293,7 +1293,7 @@ mod tests {
                 name: "x".to_string(),
                 transform: Transform::Identity,
                 source_id: 1,
-                partition_id: Some(1000),
+                field_id: Some(1000),
```

Comment:
I'm a bit surprised that the test against `testdata/table_metadata/TableMetadataV2Valid.json` passed.

Reply:
sure

Reply:
When build PartitionSpec, `UnboundPartitionField` is transformed to `PartitionField`,  in which `partition_id` is transformed `field_id` 
```rust
            bound_fields.push(PartitionField {
                source_id: field.source_id,
                field_id: partition_id,
                name: field.name,
                transform: field.transform,
            })
```
https://github.com/apache/iceberg-rust/blob/main/crates/iceberg/src/spec/partition.rs#L415-L420

Reply:
By the way, I think `add_partition_field` is more general to use, seems `add_unbound_field` should only be used in test to specify partition field id.

Reply:
Thanks, would you like to submit a new issue for disucsion. I believe this PR is good to go.

Reply:
ok, thanks